api/v1alpha1/wasm_types.go

```diff
@@ -100,31 +113,32 @@ const (
 // HTTPWasmCodeSource defines the HTTP URL containing the wasm code.
 type HTTPWasmCodeSource struct {
 	// URL is the URL containing the wasm code.
+	// +kubebuilder:validation:Pattern=`^((https?:)(\/\/\/?)([\w]*(?::[\w]*)?@)?([\d\w\.-]+)(?::(\d+))?)?([\/\\\w\.()-]*)?(?:([?][^#]*)?(#.*)?)*`
 	URL string `json:"url"`
 }
 
 // ImageWasmCodeSource defines the OCI image containing the wasm code.
 type ImageWasmCodeSource struct {
 	// URL is the URL of the OCI image.
+	// URL can be in the format of `registry/image:tag` or `registry/image@sha256:digest`.
```

Comment:
we should also mention the name and the expected file location for the module in the oci image